src/Utils/kernel_launching.jl

```diff
@@ -33,3 +34,33 @@ function launch_config(grid, dims)
         return (threads=Tuple(threads), blocks=Tuple(blocks))
     end
 end
+
+# See: https://github.com/climate-machine/Oceananigans.jl/pull/308
+function work_layout(grid, dims)
+
+    workgroup = grid.Nx == 1 ? (1, min(256, grid.Ny)) :
+                grid.Ny == 1 ? (1, min(256, grid.Nx)) : 
+                               (16, 16)
```

Comment:
Is it worth considering the column model `grid.Nx == 1 && grid.Ny == 1` case?

Reply:
I think we should. Would it just be `(1, min(256, grid.Nz))`?

Reply:
Hmmm, are work groups 3D? I guess for max efficiency wouldn't it be `(1, 1, grid.Nz)`? If they're 2D then `(1, 1)` might make sense. Column models aren't gonna be fast on the GPU anyway.

Reply:
Also, to get rid of magic number we should maybe define
```julia
const MAX_THREADS_PER_BLOCK = 256
```
then use `Int(√256)` instead of `16`.